src/main/java/service/RacingGameService.java

```diff
@@ -0,0 +1,59 @@
+package service;
+
+import domain.Car;
+
+import java.util.List;
+
+public class RacingGameService {
+    public void playRacingGame(List<Car> cars, int count) {
+        System.out.println("실행 결과");
+        printCarsMoving(cars);
+
+        for (int i = 0; i < count; i++) {
+            moveCars(cars);
+            printCarsMoving(cars);
+        }
+    }
+
+    public int getLongestMoveDistance(List<Car> cars) {
+        List<Integer> distances = cars.stream()
+                .map(Car::getMoveDistance)
+                .toList();
+
+        return distances.stream()
+                .max(Integer::compareTo)
+                .orElseThrow(() -> new IllegalArgumentException("carList is empty"));
```

Comment:
```suggestion
        return cars.stream()
                .map(Car::getMoveDistance)
                .max(Integer::compareTo)
                .orElseThrow(() -> new IllegalArgumentException("carList is empty"));
```

`List`를 생성하고 다시 `Stream`을 사용해서 최댓값을 구하는데 위와 같이 한 번의 연산으로도 처리할 수 있습니다!

이 부분 또한, 예외 메세지를 더 자세하게 적어주면 좋을 것 같아요 🤭

Reply:
리뷰 감사합니다.

stream에 익숙하지 않아 연습 겸 Java언어로 문자열을 다루는 알고리즘 문제를 풀어보려합니다. 

예외메시지 또한 더욱 더 구체적으로 의미를 전달할 수 있게 작성해보겠습니다.

감사합니다. ^^